### ISSUE TITLE
Failed to deserialize query string: invalid type map, expected tuple of size 2

### ISSUE THREAD

**Issue by scroobius-pip** [jvdwrf/axum-typed-routing#1]
Hi,
I'm using the latest version (0.1.3), i can't seem to get query parameters to work.

To reproduce, use the example in the readme:

```
use axum_typed_routing_macros::route;

#[route(GET "/item/:id?amount&offset")]
pub async fn main(id: u32, amount: Option<u32>, offset: Option<u32>) -> String {
    log::info!("{}?amount={:?}&offset={:?}", id, amount, offset);
    id.to_string()
}
```

A get request to `localhost:8000/item/40?amount=20&offset=12` gives the following error:

`Failed to deserialize query string: invalid type: map, expected a tuple of size 2`

**Comment by jvdwrf** [comment]
Hi there! Thanks for the bug report.

I have personally not used the query params, so that is why it must have slipped by. I have pushed a new version that supports axum 0.7. 

It would be nice if you could test out whether the problem still occurs with the new version :). If it looks good, then I will push a new version to crates.io.

**Comment by scroobius-pip** [comment]
Alright will do soonest, thanks alot

**Comment by jvdwrf** [comment]
I've added some tests and released 0.2

Let me know if it works for you now :)

**Comment by scroobius-pip** [comment]
It works! Thanks alot!
